bin/plugin/lib/utils.js

```diff
@@ -23,30 +24,18 @@ const { log, formats } = require( './logger' );
  * @param {string=} cwd    Working directory.
  * @param {Env=}    env    Additional environment variables to pass to the script.
  */
-function runShellScript( script, cwd, env = {} ) {
-	return new Promise( ( resolve, reject ) => {
-		childProcess.exec(
-			script,
-			{
-				cwd,
-				env: {
-					NO_CHECKS: 'true',
-					PATH: process.env.PATH,
-					HOME: process.env.HOME,
-					USER: process.env.USER,
-					...env,
-				},
-			},
-			function ( error, stdout, stderr ) {
-				if ( error ) {
-					console.log( stdout ); // Sometimes the error message is thrown via stdout.
-					console.log( stderr );
-					reject( error );
-				} else {
-					resolve( true );
-				}
-			}
-		);
+async function runShellScript( script, cwd, env = {} ) {
+	const execPromisified = promisify( childProcess.exec );
+
+	return await execPromisified( script, {
+		cwd,
+		env: {
+			NO_CHECKS: 'true',
+			PATH: process.env.PATH,
+			HOME: process.env.HOME,
+			USER: process.env.USER,
+			...env,
+		},
```

Comment:
Another thing to try is using `spawn` instead of `exec`. The difference is that you can pass the `stdio: 'inherit' option to `spawn`.

`exec` is running the command inside a shell, but we don't need that for `npm run`. There are other usages of `runShellScript` that run `cp -R`, and `cp` is a shell built-in -- for these whe have the `shell: true` option for `spawn`.

Reply:
Agreed! I was also [considering it](https://github.com/WordPress/gutenberg/pull/54784#issuecomment-1733731753) some time ago when we had similar logging issues, as it also allow for better debugging since it streams the output, which is often quite large. Never got to trying it out, though. Anyhow, if this quick-fix doesn't help, we can try refactoring to `spawn`.

Reply:
One more idea to try: run directly the `test-playwright.js` script. This works locally:
```
node ./packages/scripts/scripts/test-playwright.js --config test/performance/playwright.config.ts site-editor
```
Currently there are so many levels of indirection. The GitHub action spawns a `node bin/plugin/cli.js` process, which spawns a `npm run` process, which spawns a `wp-scripts` process, which spawns a `test-playwright` process, which finally spawns the `playwright` CLI process. I counted five processes 🙂 Somewhere along the way the errors get lost.

Reply:
It's still the same, unfortunately. 😩 I'll keep digging, but if you have any other ideas, please share! 😄